src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,28 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
 
-export const TodoModal: React.FC = () => {
+type Props = {
+  todo: Todo;
+  setIsOpen: (isOpen: boolean) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ todo, setIsOpen }) => {
+  const [user, setUser] = useState<User | null>(null);
+
+  getUser(todo.userId).then(setUser);
```

Comment:
`useEffect` is like a helper that tells React to do something after the component has rendered. In this case, it’s helping us fetch user data from an API when the `todo.userId` changes. Without `useEffect`, the fetch would run every time the component renders, which is not what we want.

Here’s a simpler version of your code with `useEffect`:

```jsx
import React, { useEffect, useState } from 'react';

const MyComponent = ({ todo }) => {
  const [user, setUser] = useState(null);

  useEffect(() => {
    if (todo.userId) {
      getUser(todo.userId).then(setUser);
    }
  }, [todo.userId]);

  return (
    <div>
      {user ? <p>User: {user.name}</p> : <p>Loading...</p>}
    </div>
  );
};

export default MyComponent;
```

### Simplified Explanation
1. **Purpose**: `useEffect` runs code after the component is first rendered and whenever certain values (like `todo.userId`) change.
2. **When to Fetch**: We only want to fetch user data when the `todo.userId` changes. That’s why we put the fetch code inside `useEffect` and add `[todo.userId]` as a dependency.
3. **Preventing Extra Work**: Without `useEffect`, the fetch would run on every render, which is wasteful and could cause problems.

So, by using `useEffect`, we make sure we fetch user data only when we need to, making our component efficient and avoiding unnecessary work.